examples/python/debug/test/_debug_example_test.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test for gRPC Python debug example."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import unittest
+from contextlib import contextmanager
+import socket
+
+from examples.python.debug import debug_server
+from examples.python.debug import send_message
+from examples.python.debug import get_stats
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+_FAILURE_RATE = 0.5
+_NUMBER_OF_MESSAGES = 100
+
+
+@contextmanager
+def get_free_loopback_tcp_port():
```

Comment:
I'm not sure of the best way to deal with this, but I've noticed that we've duplicated this code a lot in both example code and test code. I wish we didn't have to copy-paste every time. I'm also not thrilled about including it in our API surface. Thoughts? Maybe we could create an entirely separate utility library for it?

Reply:
Alternatively, we can talk to our local port server. But it wouldn't make any sense to people who doesn't know about it.

What kind of utility library are you talking about? 1) Like grpc-* packages? 2) Or a completely separate Python package; 3) Or a common utility module inside `grpcio-tests` that only we have access to?